src/tir/transforms/lower_thread_allreduce.cc

```diff
@@ -33,10 +33,33 @@
 
 #include "../../runtime/thread_storage_scope.h"
 #include "ir_utils.h"
+#include "update_pointer_storage_scope.h"
 
 namespace tvm {
 namespace tir {
 
+class UpdatePointerStorageScopeAllReduce final : public UpdatePointerStorageScope {
```

Comment:
Can we merge this pass with `UpdatePointerStorageScope` by adding check for `attr::volatile_scope` in `VisitStmt_(const AttrStmtNode*)`

Reply:
This `attr::volatile_scope` is created inside this `AllocateNode` visitor. A visitor for `AttrStmtNode` don't get to see this.